api/src/main/java/org/openmrs/module/sync2/api/utils/SyncUtils.java

```diff
@@ -1,188 +1,180 @@
+        }
+        return map.get(ATOMFEED_TAG_VALUE_FIELD_NAME);
+    }
+    
+    public static String selectAppropriateClientName(Map<String, String> availableResourceLinks) {
+        String preferredClient = getPreferredClient();
+        if (availableResourceLinks.containsKey(preferredClient)) {
+            return preferredClient;
+        } else {
+            return getFirstKey(availableResourceLinks);
+        }
+    }
+    
+    private static String getFirstKey(Map<String, String> map) {
+        if (map.size() > 0) {
+            return map.keySet().iterator().next();
+        } else {
+            throw new SyncException("A map doesn't contain any entries, so it is impossible to get the first key");
+        }
+    }
+
+    public static String getPreferredClient() {
+        return Context.getAdministrationService().getGlobalProperty(RESOURCE_PREFERRED_CLIENT);
+    }
+
+    public static String extractUUIDFromResourceLinks(Map<String, String> resourceLinks) {
+        for (String client : resourceLinks.keySet()) {
+            switch (client) {
+                case REST_CLIENT:
+                    return extractUUIDFromRestResource(resourceLinks.get(client));
+                case FHIR_CLIENT:
+                    return extractUUIDFromFHIRResource(resourceLinks.get(client));
+                default:
+            }
+        }
+
+        LOGGER.error("Couldn't find any supported client to extract uuid from.");
+        return null;
+    }
+
+    public static void readFeedByCategory(String category, AtomFeedClient atomFeedClient,
+                                          SyncConfigurationService configurationService, String ws) {
+        try {
+            URI uri = new URI(getResourceUrlWithCategory(category, configurationService, ws));
+            atomFeedClient.setUri(uri);
+            atomFeedClient.process();
+        } catch (URISyntaxException e) {
+            throw new SyncException("Atomfeed URI is not correct. ", e);
+        } catch (Exception e) {
+            throw new SyncException(String.format("Error during processing atomfeeds for category %s: ", category), e);
+        }
+    }
+
+    //TODO add handling Visits, Encounters and Obs
+    public static boolean compareLocalAndPulled(String clientName, String category, Object from, Object dest) {
+        boolean result = false;
+        if (null != dest && null != from) {
+            //If 'from' is  instance of String it represent uuid and should be used to delete object action.
+            if (!(from instanceof String)) {
+                switch (clientName) {
+                    case REST_CLIENT:
+                    	if(category.equals(SyncCategoryConstants.CATEGORY_AUDIT_MESSAGE)) {
+                    		result = ((AuditMessage) from).getUuid().equals( ((AuditMessage) dest).getUuid());
+                    	} else {
+                    		result = ((SimpleObject) from).get("uuid").equals(((SimpleObject) dest).get("uuid"));
```

Comment:
You have to compare all essential fields to check if the object is updated/edited. If it is then it has to be pulled.
Fields like "links" can't be compared - they are different even if the object is equal. So simple from.equals(dest) most probably won't work.